src/SOS/Strike/strike.cpp

```diff
@@ -9244,8 +9285,194 @@ DECLARE_API(u)
         }
     }
 
+    ExtractionCodeHeaderResult p = extractCodeHeaderData(methodDesc, dwStartAddr);
+    Status = std::get<2>(p);
+    if (Status != S_OK)
+    {
+        return Status;
+    }
+
+    NameForMD_s(methodDesc, g_mdName, mdNameLen);
+    ExtOut("%S\n", g_mdName);
+
+    DacpMethodDescData& MethodDescData = std::get<0>(p);
+    DacpCodeHeaderData& codeHeaderData = std::get<1>(p);
+    std::unique_ptr<CLRDATA_IL_ADDRESS_MAP[]> map(nullptr);
+    ULONG32 mapCount = 0;
+    Status = displayIntermediateLanguage(bIL, codeHeaderData, map /*out*/, mapCount /* out */);
+    if (Status != S_OK)
+    {
+        return Status;
+    }
+
+    // ///////////////////////////////////////////////////////////////////////////
+    // This can be reused with sildasm but kept as-is largely since it just
+    // works so it can be fixed later.
+    // ///////////////////////////////////////////////////////////////////////////
+
+    if (MethodDescData.bIsDynamic && MethodDescData.managedDynamicMethodObject)
+    {
+        ExtOut("Can only work with dynamic not implemented\n");
+        return Status;
+    }
+
+    GetILAddressResult result = GetILAddress(MethodDescData);
+    if (std::get<0>(result) == NULL)
+    {
+        ExtOut("ilAddr is %p\n", SOS_PTR(std::get<0>(result)));
+        return E_FAIL;
+    }
+    ExtOut("ilAddr is %p pImport is %p\n", SOS_PTR(std::get<0>(result)), SOS_PTR(std::get<1>(result).get()));
+    TADDR ilAddr = std::get<0>(result);
+    std::shared_ptr<IMetaDataImport> pImport(std::move(std::get<1>(result)));
+    /// Taken from DecodeILFromAddress(IMetaDataImport *pImport, TADDR ilAddr)
+    ULONG Size = GetILSize(ilAddr);
+    if (Size == 0)
+    {
+        ExtOut("error decoding IL\n");
+        return Status;
+    }
+    // Read the memory into a local buffer
+    ArrayHolder<BYTE> pArray = new BYTE[Size];
+    Status = g_ExtData->ReadVirtual(TO_CDADDR(ilAddr), pArray, Size, NULL);
+    if (Status != S_OK)
+    {
+        ExtOut("Failed to read memory\n");
+        return Status;
+    }
+    /// Taken from DecodeIL(pImport, pArray, Size);
+    // First decode the header
+    BYTE *buffer = pArray;
+    ULONG bufSize = Size;
+    COR_ILMETHOD *pHeader = (COR_ILMETHOD *) buffer;
```

Comment:
Build fails on CentOS and MacOSX. There is a some header file missing or maybe this is specific to Windows operating system. Need to dig more on this.
```
[ 97%] Building CXX object src/SOS/Strike/CMakeFiles/sos.dir/strike.cpp.o
/home/vsts/work/1/s/src/SOS/Strike/strike.cpp:9352:5: error: unknown type name 'COR_ILMETHOD'
 COR_ILMETHOD *pHeader = (COR_ILMETHOD *) buffer;
 ^
/home/vsts/work/1/s/src/SOS/Strike/strike.cpp:9352:30: error: use of undeclared identifier 'COR_ILMETHOD'
 COR_ILMETHOD *pHeader = (COR_ILMETHOD *) buffer;
 ^
/home/vsts/work/1/s/src/SOS/Strike/strike.cpp:9352:44: error: expected expression
 COR_ILMETHOD *pHeader = (COR_ILMETHOD *) buffer;
 ^
/home/vsts/work/1/s/src/SOS/Strike/strike.cpp:9353:5: error: unknown type name 'COR_ILMETHOD_DECODER'
 COR_ILMETHOD_DECODER header(pHeader);
 ^
4 errors generated.
```